iree/compiler/Dialect/HAL/Target/LLVM/internal/WasmLinkerTool.cpp

```diff
@@ -50,8 +50,9 @@ class WasmLinkerTool : public LinkerTool {
       return std::string(linkerPath);
     }
 
-    // No explicit linker specified, search the environment for common tools.
-    std::string toolPath = findToolInEnvironment(
+    // No explicit linker specified, search the environment (i.e. our own build
+    // or install directories) for common tools.
+    const std::string &toolPath = findToolFromExecutableDir(
```

Comment:
const& seems wrong here, given that the function is returning a value? (this is a good place for auto)

Reply:
Why wrong? It's fine to bind a const reference to a temporary. That extends the temporary's lifetime accordingly so there's no gotcha here. That's a traditional way to use returned-by-value objects without an unnecessary copy or move or other magic to elide such work.

Reply:
Note though, the `const` is required here, it's not just const-correct pedantry. Because why C++ allows binding a const-reference to a temporary, it doesn't allow binding a non-const-reference to a temporary.